app/subscriber/src/features/filter-media/FilterMediaLanding.tsx

```diff
@@ -106,14 +135,92 @@ export const FilterMediaLanding: React.FC = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [activeLetter, activeFilter, filter]);
 
-  // a little bit funky, but needed to keep track if the select all option is checked to maintain checked state
-  const allSelected = {
-    sourceIds: narrowedOptions
-      .filter((opt) => opt.listOption === ListOptionName.Source)
-      .map((opt) => opt.id),
-    seriesIds: narrowedOptions
-      .filter((opt) => opt.listOption === ListOptionName.Series)
-      .map((opt) => opt.id),
+  const checkAllOptions = (filter: ISubMediaGroupItem, value: boolean) => {
+    filter?.options.forEach((x) => {
+      x.selected = value;
+    });
+  };
+
+  const isFullSelected = (filter: ISubMediaGroupItem | undefined) => {
+    if (filter) {
+      return filter?.options.filter((x) => x.selected === false).length === 0;
+    }
+    return false;
+  };
+
+  const handleGroupClick = (mediaGroup: ISubMediaGroupItem) => {
+    if (mediaGroup.label === 'Weekly Print') setActiveLetter('A');
+    else if (mediaGroup.label === 'Online') setActiveLetter('B');
+    else setActiveLetter('All');
+    checkAllOptions(mediaGroup, true);
+    let sourceIds = mediaGroup.options
+      .filter((x) => x.listOption === ListOptionName.Source && x.selected === true)
+      .map((x) => x.id);
+    let seriesIds = mediaGroup.options
+      .filter((x) => x.listOption === ListOptionName.Series && x.selected === true)
+      .map((x) => x.id);
+    let emptyArrayTerms: string[] = [];
+    if (mediaGroup?.listOption === ListOptionName.Series) {
+      emptyArrayTerms = ['seriesId'];
+    }
+    if (mediaGroup?.listOption === ListOptionName.Source) {
+      emptyArrayTerms = ['sourceId'];
+    }
+    storeFilter({
+      ...filter,
+      sourceIds,
+      seriesIds,
+      emptyArrayTerms,
+      mediaTypeIds: [mediaGroup.key],
+    });
+    setActiveFilter(mediaGroup);
+    setParentClicked(true);
+  };
+
+  const handleClickAll = (e: any) => {
```

Comment:
Should any of these helper functions become `React.useCallback(...)`?  Currently they would result in a new function every time the code is traversed.